core/src/main/scala/cats/data/Prod.scala

```diff
@@ -22,13 +23,23 @@ private[data] sealed abstract class ProdInstances extends ProdInstances0 {
     def eqv(x: Prod[F, G, A], y: Prod[F, G, A]): Boolean =
       FF.eqv(x.first, y.first) && GG.eqv(x.second, y.second)
   }
+
+  implicit def catsDataShowForProd[F[_], G[_], A](implicit FF: Show[F[A]], GF: Show[G[A]]): Show[Prod[F, G, A]] = new ProdShow[F, G, A] {
+    def F: Show[F[A]] = FF
+    def G: Show[G[A]] = GF
+  }
 }
 
 private[data] sealed abstract class ProdInstances0 extends ProdInstances1 {
   implicit def catsDataMonoidKForProd[F[_], G[_]](implicit FF: MonoidK[F], GG: MonoidK[G]): MonoidK[λ[α => Prod[F, G, α]]] = new ProdMonoidK[F, G] {
     def F: MonoidK[F] = FF
     def G: MonoidK[G] = GG
   }
+
+  implicit def catsDataOrderForProd[F[_], G[_], A](implicit FF: Order[F[A]], GF: Order[G[A]]): Order[Prod[F, G, A]] = new ProdOrder[F, G, A] {
```

Comment:
As @adelbertc pointed out, `Order` instance should have higher priority than `Eq` since it's more specific. I think we should move this instance to `ProdInstances` and the `Eq` instance down here.